lib/rack/mock_response.rb

```diff
@@ -78,7 +78,7 @@ def cookie(name)
 
     def parse_cookies_from_header
       cookies = Hash.new
-      if headers.has_key? 'set-cookie'
+      if headers.has_key?('set-cookie') && !headers.fetch('set-cookie').empty?
         set_cookie_header = headers.fetch('set-cookie')
```

Comment:
```suggestion
      set_cookie_header = headers['set-cookie']
      if set_cookie_header&.any?
```

How about this? To avoid looking up the hash multiple times?